examples/nlp/text_generation_gpt.py

```diff
@@ -0,0 +1,418 @@
+"""
+Title: Simple GPT Text Generation with KerasNLP transformers
+Author: [Jesse Chan](https://github.com/jessechancy)
+Date created: 2022/07/07
+Last modified: 2022/07/07
+Description: Use KerasNLP transformers to train a mini-GPT model for text generation.
+"""
+
+"""
+## Introduction
+
+In this example, we will use KerasNLP layers to build a scaled down Generative
+Pre-trained (GPT) model. GPT is a transformer based model that allows you to generate
+sophisticated text from a small input.
+
+We will train the model on the [simplebooks-92](https://arxiv.org/abs/1911.12391) corpus,
+which is a dataset made from several novels. It is a good dataset for this example since
+it has a small vocabulary and high word frequency, which is beneficial when training a
+model with few parameters.
+
+This example combines concepts from [Text generation with a miniature GPT](https://keras.io/examples/generative/text_generation_with_miniature_gpt/) 
+with KerasNLP abstractions. We will demonstrate how KerasNLP tokenization, model, metrics, and
```

Comment:
Maybe just:
We will demonstrate how KerasNLP tokenization, layers and metrics simplify the training
process, and then show how to generate output text using sampling utilities.

And then remove the whole next paragraph. Readers can read on to see the exact layers used.

Reply:
edited